File: Bubble-Sort/BS.hs
{--main = do
  let testData = [14,33,27,35,10]
  let flag = True
  print (bubbleSort testData flag)
--}
bubbleSort :: ([Integer],Bool) -> ([Integer],Bool)
bubbleSort ((x:[]),a) = ([x],a)
bubbleSort (y:x:xs)
  | x < y = x:(bubbleSort (y:xs))
  | otherwise = y:(bubbleSort (x:xs))

File: Bubble-Sort/BS.jl
function bubbleSort(ll)
    flag = true
    len = length(ll)
    while flag
        flag = false
        this = ll[1]
        result = [];
        for i in 2:len
            if ll[i] < this
                push!(result, ll[i])
                flag = true
            else
                push!(result, this)
                this = ll[i];
            end
        end
        push!(result,this)
        ll = result
    end
    ll
end

using Test
@test bubbleSort([14,33,27,35,10]) == [10,14,27,33,35]
